src/System.Private.CoreLib/System.Private.CoreLib.csproj

```diff
@@ -510,10 +510,13 @@
     <Compile Include="$(BclSourcesRoot)\System\Diagnostics\Eventing\EventSource_CoreCLR.cs" />
     <Compile Condition="'$(FeatureXplatEventSource)' == 'true'" Include="$(BclSourcesRoot)\System\Diagnostics\Eventing\XplatEventLogger.cs" />
     <Compile Include="$(BclSourcesRoot)\System\Diagnostics\Eventing\FrameworkEventSource.cs" />
+    <Compile Condition="Exists('$(IntermediateOutputPath)..\eventing\DotNETRuntimeEventSource.cs')" Include="$(IntermediateOutputPath)..\eventing\DotNETRuntimeEventSource.cs" />
```

Comment:
Have you looked into why these files do not exist in some build configurations? I do not think that this Condition is going to work now that this is starting to do something useful.

Reply:
I have looked into this at a high-level.  From what I can remember there are some build steps that cause System.Private.CoreLib to be built even though that's not the intention.  At least one of these occurs before DotNETRuntimeEventSource.cs has been generated - thus the error.

I suspect that it's "OK" in that right now it will work, but I do think that it's really not ideal and is likely to cause problems at the worst possible time.  I have filed #18661 to track this.

Reply:
@jkotas, FYI, I have resolved #18661 in a separate PR.